### PR TITLE
[PF-2343] Make the max retry time greater than the socket timeout

### PR DESCRIPTION
**Pull request by cbookg**

### CODE REVIEW

**Review by melissachang — APPROVED** [review]
Nice

File: service/src/main/java/bio/terra/workspace/common/utils/RetryRules.java [hunk 1]
```diff
@@ -31,7 +31,7 @@ public static RetryRule cloudLongRunning() {
    * few seconds.
    */
   public static RetryRule shortExponential() {
-    return new RetryRuleExponentialBackoff(1, 8, 16);
```
[the remark this answers is not in the document]
Comment (ddietterich) [reply service/src/main/java/bio/terra/workspace/common/utils/RetryRules.java:34]: I don't have a problem with using 30secs, but I do not follow @melissachang logic.

This retry is the time to wait _between_ failures. So a step would run for however long it runs. Maybe it fails with a socket timeout and that is retried. Only then does this wait happen. Maybe I'm missing something, but I don't see how the retry wait relates to the socket timeout here.

**Review by ddietterich — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/common/utils/RetryRules.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbookg) [reply service/src/main/java/bio/terra/workspace/common/utils/RetryRules.java:34]: The first value is the time to wait between failures.  The 2nd value is the max this "wait between failures" should be increased.

The 3rd value is the "max overall operation time", where retrying will be stopped after this end time:
https://github.com/DataBiosphere/stairway/blob/38347550357a85d0a91c1218289ac5a53a1356e9/stairway/src/main/java/bio/terra/stairway/RetryRuleExponentialBackoff.java#L42

[the remark this answers is not in the document]
Comment (melissachang) [reply service/src/main/java/bio/terra/workspace/common/utils/RetryRules.java:34]: My understanding [from the logs](https://cloudlogging.app.goo.gl/yXKiGLbBib6WuGLX9):

- 01:22:35 - GrantWsmRoleAdminStep started
- 01:22:55 - GrantWsmRoleAdminStep failed due to socket timeout
- 01:22:55.891 PST

ws-manager
Retry rule exponential: now (2022-12-08T09:22:55.891792730Z) is past max time 2022-12-08T09:22:51.849920864Z - not retrying

[the remark this answers is not in the document]
Comment (ddietterich) [reply service/src/main/java/bio/terra/workspace/common/utils/RetryRules.java:34]: Ah, OK. Is that the semantic you want to use? We can code a different retry if we like.

**Pull request merged**